File: Backend/Models/Entities/Vendor_Asset.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Vendor_Asset
    {
        [Key]
        public int Vendor_Asset_ID { get; set; }

        [Required]
        public int Asset_ID { get; set; }
        [Required]
        public int Vendor_ID { get; set; }
        [ForeignKey("Asset_ID")]
        public Asset Asset { get; set; }
        [ForeignKey("Vendor_ID")]
        public Vendor Vendor { get; set; }
    }
}

File: Frontend/src/app/Shared/Vendor_Asset.ts
import { Asset } from "./Asset";
import { VendorOnboardRequest } from "./VendorOnboardRequest";

export interface Vendor_Asset {
    vendor_Asset_ID:number;
    asset_ID:number;
    vendor_ID:number;
    asset:Asset;
    vendor:VendorOnboardRequest;
}